src/main/java/com/greedobank/cards/controller/CardController.java

```diff
@@ -0,0 +1,57 @@
+package com.greedobank.cards.controller;
+
+import com.greedobank.cards.dto.CardCreationDTO;
+import com.greedobank.cards.dto.CardDTO;
+import com.greedobank.cards.dto.CardResetPinDTO;
+import com.greedobank.cards.service.CardService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+@RestController
+@Validated
+@RequestMapping(path = "/api/v1/card")
+@Tag(name = "Card", description = "Interaction with cards")
+public class CardController {
+    private final CardService cardService;
+
+    @Autowired
+    public CardController(CardService cardService) {
+        this.cardService = cardService;
+    }
+
+    @PostMapping
+    @Operation(summary = "Create a card", description = "Lets create a card")
+    @PreAuthorize("hasRole('ADMIN')")
+    public CardDTO create(@Valid @RequestBody CardCreationDTO cardCreationDTO) {
+        return cardService.create(cardCreationDTO);
+    }
+
+    @GetMapping(value = "/{id}")
+    @Operation(summary = "Get the card by id", description = "Lets get the card by id")
+    @PreAuthorize("isAuthenticated()")
+    public CardDTO getById(@PathVariable("id") int id) {
+        return cardService.getById(id);
+    }
+
+    @PatchMapping(value = "/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
```

Comment:
204 (NO CONTENT) is not that recommended for service to return, in case of some errors, we'd better provide the client with some exception description and an internal error response code (5xx)

Reply:
I use here 204 status because it indicates that the server has successfully fulfilled the request and that there is no content to send in the response payload body.  And when I tried to add handle exceptions for 5xx status, I couldn't get appropriate messages for specific exceptions described earlier: 
    @ExceptionHandler(Exception.class)
    @ResponseStatus(value = HttpStatus.INTERNAL_SERVER_ERROR)
    @ResponseBody
    public JsonErrorResponse handleServiceUnavailable() {
        return new JsonErrorResponse(SOMETHING_WENT_WRONG.getDescription());
    }

Reply:
when you use @ExceptionHandler (or ControllerAdvice), you're free to specify any concrete/abstract exception within inheritance hierarchy